var/spack/repos/builtin/packages/jdk/package.py

```diff
@@ -58,6 +58,9 @@ class Jdk(Package):
     # away you go.
     version('7u80-b0', '6152f8a7561acf795ca4701daa10a965')
 
+    provides('java@8', when='@8u66-b17' '@8u73-b02' '@8u92-b14' '@8u131-b11')
+    provides('java@7', when='@7u80-b0')
```

Comment:
I don't think this works? Why not say:
```python
provides('java@8', when='@8u0:8u999')
provides('java@7', when='@7u0:7u999')
```
You can confirm that this works like so:
```
$ spack python
Spack version 0.10.0
Python 2.7.13, Darwin x86_64
>>> from spack.version import *
>>> Version('8u66-b17') > Version('8u0')
True
>>> Version('8u66-b17') < Version('8u999')
True
```